src/api_server/src/lib.rs

```diff
@@ -200,6 +200,11 @@ impl ApiServer {
             _ => None,
         };
 
+        let vmm_new_state = match *vmm_action {
+            VmmAction::Pause => "Paused".to_string(),
+            VmmAction::Resume => "Running".to_string(),
+            _ => self.vmm_shared_info.read().unwrap().state.clone(),
+        };
```

Comment:
Instead of using intermediate variable `vmm_new_state`, you can move this block of code lower under

```rust
if vmm_outcome.is_ok() {
    // change state here depending on what VmmAction succeeded
}
```

Reply:
Yes, I tried that initially but it would not work since vmm_action is moved when we call self.api_request_sender(vmm_action). I think the current implementation might involve less change to the existing design.